specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json

```diff
@@ -667,25 +675,124 @@
-            "EndsWith",
-            "EqualsAny"
-          ],
           "description": "When matchVariable is a collection, operate on the selector to specify which elements in the collection this exclusion applies to.",
-          "x-ms-enum": {
-            "name": "OwaspCrsExclusionEntrySelectorMatchOperator",
-            "modelAsString": true
-          }
+          "$ref": "#/definitions/MatchOperator"
         },
         "selector": {
           "type": "string",
           "description": "When matchVariable is a collection, operator used to specify which elements in the collection this exclusion applies to."
         }
       }
+    },
+    "MatchOperator": {
+      "type": "string",
+      "enum": [
+        "Equals",
+        "Contains",
+        "StartsWith",
+        "EndsWith",
+        "EqualsAny"
+      ],
+      "description": "Defines the logic for comparing two strings.",
+      "x-ms-enum": {
+        "name": "OwaspCrsExclusionEntrySelectorMatchOperator",
+        "modelAsString": true
+      }
+    },
+    "BotDetection": {
+      "description": "Allows the detection of bots and specify action for each bot category.",
+      "required": [
+        "good", 
+        "neutral", 
+        "bad"
+      ],
+      "properties": {
+        "good": {
+          "description": "Describes the action to be taken when detecting a good bot.",
+          "properties": {
+            "action": {
+              "description": "Describes which action should be triggered by the bot category.",
+              "$ref": "#/definitions/Action"
+            }
+          }
+        },
+        "neutral": {
+          "description": "Describes the action to be taken when detecting a neutral bot.",
+          "properties": {
+            "action": {
+              "description": "Describes which action should be triggered by the bot category.",
+              "$ref": "#/definitions/Action"
+            }
+          }
+        },
+        "bad": {
+          "description": "Describes the action to be taken when detecting a bad bot.",
+          "properties": {
+            "action": {
+              "description": "Describes which action should be triggered by the bot category.",
+              "$ref": "#/definitions/Action"
+            }
+          }
+        },
+        "overrides": {
+          "description": "List of custom bot categories which take precedence over the predefined categories.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CustomBotCategory"
+          }
+        }
+      }
+    },
+    "BotIdentifier": {
+      "description": "Defines a way to capture a bot request.",
+      "required": [
+        "type",
+        "value",
+        "matchOperator"
+      ],
+      "properties": {
+        "type": {
+          "description": "Describes which property of the request to match against.",
+          "type": "string",
+          "enum": [
+            "IP Address",
+            "MAC Address"
```

Comment:
I don't think MAC address makes any sense here. That's a layer 2 concept. The MAC address of all IP packets reaching our VM's will be the MAC address of some router in our data center.

Reply:
Oh okay, I did not know that. good to know!